### PR TITLE
Fix for StorageManager null bug

### PR DESCRIPTION
**Pull request by Hackmodford**
Added null check to verifySpace(int destination, String path, long length)

My last pull request introduced a new bug. It was possible that there is a null File within the array. The new code checks for this.

### CODE REVIEW

**Comment by stefanhoth** [comment]
Can one of the admins review this PR?

**Comment by frapontillo** [comment]
CI, test this please.

File: library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java [hunk 1]
```diff
@@ -179,11 +179,14 @@ void verifySpace(int destination, String path, long length) throws StopRequestEx
...
                 } else if (path.startsWith(internalStorageDir.getPath())) {
                     dir = internalStorageDir;
-                }
-                for (File aExternalStorageDir : externalStorageDirs) {
-                    if (path.startsWith(aExternalStorageDir.getPath())) {
-                        dir = aExternalStorageDir;
-                        break;
+                } else {
+                    for (File aExternalStorageDir : externalStorageDirs) {
+                        if (aExternalStorageDir != null) {
```
Comment (frapontillo) [inline library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java]: Can you collapse both `if`s into one?

**Comment by Hackmodford** [comment]
I created the change as requested.

File: library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java [hunk 2]
```diff
@@ -179,11 +179,12 @@ void verifySpace(int destination, String path, long length) throws StopRequestEx
                     dir = systemCacheDir;
                 } else if (path.startsWith(internalStorageDir.getPath())) {
                     dir = internalStorageDir;
-                }
-                for (File aExternalStorageDir : externalStorageDirs) {
-                    if (path.startsWith(aExternalStorageDir.getPath())) {
-                        dir = aExternalStorageDir;
-                        break;
+                } else {
+                    for (File aExternalStorageDir : externalStorageDirs) {
```
Comment (Hackmodford) [inline library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java]: Fixed it. Thanks ;)

**Comment by takecare** [comment]
thanks @Hackmodford

**Pull request merged**